### PR TITLE
Sound effects reference pages

### LINKED ISSUE #4516: No help page for block "play sound effect"
**Describe the bug**
No help doc for block "**play sound**"

**Steps to reproduce the behavior**
1.Nabigate to https://makecode.microbit.org/beta#editor
2.Drag the block "**play sound**" from **Music**
3.Right-click on it and choose **'Help'**

**Expect behavior**
Show the help doc for block "**play sound**"

**Actual behavior**
No help doc for block "**play sound**"

**Additional context**
1.OS: Windows11
2.makecode.microbit.org version: 4.1.21
3.Microsoft MakeCode version: 7.5.13

### PR DESCRIPTION
**Pull request by ganicke**
Add some reference pages for the new sound effects API. I'm using the term "sound expression" to describe the set of parameters that define a sound effect.

The "info" page is for the sound editor is currently set as `./types/sound` for a placeholder. It could stay as a reference for the `Sound` type or go into a new `./info` doc folder... `./info/sound-editor.md`.

Closes #4516

### CODE REVIEW

File: docs/reference/music/create-sound-effect.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## Parameters
+
+* **waveShape**: the primary shape of the waveform:
+>* `sine`: sine wave shape
+>* `sawtooth`: sawtooth wave shape
+>* `triangle`: triangle wave shape
+>* `square`: square wave shape
+>* `noise`: random noise generated wave shape
+* **startFrequency**: a [number](/types/number) that is the frequency of the waveform when the sound expression starts.
+* **endFrequency**: a [number](/types/number) the freqeunce of the waveform when the sound expression stops.
```
Comment (riknoll) [inline docs/reference/music/create-sound-effect.md:20]:
```suggestion
* **endFrequency**: a [number](/types/number) that is the frequency of the waveform when the sound expression stops.
```

File: docs/reference/music/create-sound-effect.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* **startVolume**: a [number](/types/number) the initial volume of the sound expression.
+* **endVolume**: a [number](/types/number) the ending volume of the sound expression.
+* **duration**: a [number](/types/number) the duration in milliseconds of the sound expression.
+* **effect**: an effect to add to the waveform. These are:
+>* `tremolo`: add slight changes in volume of the sound expression.
+>* `vibrato`: add slight changes in frequency to the sound expression.
+>* `warble`: similar to `vibrato` but with faster variations in the frequency changes.
+* **interpolation**: controls the rate of frequency change in the sound expression.
+>* `linear`: the change in frequency is constant for the duration of the sound.
+>* `curve`: the change in freequency is faster at the beginning of the sound and slows toward the end.
```
Comment (riknoll) [inline docs/reference/music/create-sound-effect.md:30]:
```suggestion
>* `curve`: the change in frequency is faster at the beginning of the sound and slows toward the end.
```

**Review by Jaqster — APPROVED** [review]
Thanks so much for doing this!  Looks great, but should have Richard or @jaustin take a look for correctness.

File: docs/reference/music/create-sound-effect.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+>* `logrithmic`: the change in frequency is rapid during the very first part of the sound.
```
Comment (riknoll) [inline docs/reference/music/create-sound-effect.md:31]:
```suggestion
>* `logarithmic`: the change in frequency is rapid during the very first part of the sound.
```

File: docs/reference/music/play-sound-effect.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+![works with micro:bit V2 only image](/static/v2/v2-only.png)
+
+This block requires the [micro:bit V2](/device/v2) hardware. If you use this block with a micro:bit v1 board, you will see the **927** error code on the screen.
+
+### ~
+
+## Parameters
+
+* **sound**: a [string]() that is the sound expression for the sound you want to play.
```
Comment (riknoll) [inline docs/reference/music/play-sound-effect.md:25]:
```suggestion
* **sound**: a [string](/types/string) that is the sound expression for the sound you want to play.
```

File: docs/reference/music/play-sound-effect.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+* **mode**: the play mode for the sound, either `until done` or `background`.
+
+## Example
+
+Play a sound from a sound espression for `1` second.
```
Comment (riknoll) [inline docs/reference/music/play-sound-effect.md:30]:
```suggestion
Play a sound from a sound expression for `1` second.
```

File: docs/types/sound.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
+# Sound
+
+A **Sound** is a data object that is created from a sound expression. A sound expression is group of parameters that define a sound sound, such as wave shape, sound volume, frequency, and duration.
```
Comment (riknoll) [inline docs/types/sound.md:3]:
```suggestion
A **Sound** is a data object that is created from a sound expression. A sound expression is group of parameters that define a sound, such as wave shape, sound volume, frequency, and duration.
```

File: docs/types/sound.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+### Square wave
+
+A square wave has both verical rising and falling edges with a flat section on the top and bottom. The flat sections match the volume set for the sound. Square waves are sometimes used to represent digital data and will make an "electronic" sound.
+
+![Square wave](/static/types/sound/square-wave.png)
+
+### Noise wave
+
+The noise wave is created using random frequenices and volume. Setting the frequency parameters for the sound expression has no effect on sounds created with the noise wave.
```
Comment (riknoll) [inline docs/types/sound.md:62]: It does actually have an effect! The noise isn't true noise, but really a "tuned" noise waveform. You can get some interesting results if you set the start and end frequencies to be the same or very close to each other

File: docs/types/sound.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+* **Linear**: The change in frequency is constant for the duration of the sound.
+
+>![Frequency from low to high](/static/types/sound/interp-linear.png)
+
+* **Curve**: The change in frequency is faster at the beginning of the sound and slows toward the end.
+
+>![Frequency from low to high](/static/types/sound/interp-curve.png)
+
+* **Logrithmic**: The change in frequency is rapid during the very first part of the sound.
```
Comment (riknoll) [inline docs/types/sound.md:128]:
```suggestion
* **Logarithmic**: The change in frequency is rapid during the very first part of the sound.
```

**Review by riknoll — APPROVED** [review]

**Pull request merged**